event/main_listener.php

```diff
@@ -406,4 +413,40 @@ public function quote_text_upon_pm($event)
 			}
 		}
 	}
+
+	/**
+	 * Display contribution count and link to Titania profile in memberlist.php.
+	 * @param $event
+	 */
+	public function add_memberlist_template_vars($event)
+	{
+		if (!defined('TITANIA_CONTRIBS_TABLE'))
+		{
+			// Include Titania so we can access the constants
+			require($this->ext_root_path . 'common.' . $this->php_ext);
+		}
+
+		$this->language->add_lang(['memberlist'], 'phpbb/titania');
+
+		$sql = 'SELECT COUNT(contrib_id) as contribs
+		    FROM ' . TITANIA_CONTRIBS_TABLE . '
+		    WHERE contrib_user_id = ' . (int) $event['member']['user_id'] . '
+		      AND contrib_status = 2';
+
+		$result = $this->db->sql_query($sql);
+
+		$u_total_contribs = (int) $this->db->sql_fetchfield('contribs');
+
+		$db->sql_freeresult($result);
```

Comment:
Probably could have used the `memberlist_modify_viewprofile_sql` event instead of adding another query. Might not be a huge deal, but something to keep in mind if we start seeing problems.

Reply:
I can always push an update if it's causing problems down the road 👍 

As it's on the user profile page and only querying one user I'm confident it shouldn't be an issue on load times.